db2i_file.cc

```diff
@@ -387,9 +387,9 @@ void db2i_table::filenameToTablename(const char* in, char* out, size_t outlen)
     strcat(out, "#P#");
     accumLen += 4;
     
-    memset(temp, 0, min(outlen, part2-part1));
-    memcpy(temp, part3, min(outlen, part4-part3));
-    temp[min(outlen-1, part4-part3)] = 0;
+    memset(temp, 0, std::min(outlen, static_cast<size_t>(part2-part1)));
+    memcpy(temp, part3, std::min(outlen, static_cast<size_t>(part4-part3)));
+    temp[std::min(static_cast<size_t>(outlen-1), static_cast<size_t>(part4-part3))] = 0;
```

Comment:
I think a temp variable would be useful here. Also, looks like maybe the original code had a bug since the memset references part1 and part2, but the other two lines use part3 and part4.

Reply:
Created temp varialble of `num_bytes2` to store `static_cast<size_t>(part4-part3)`
Yes looks like bug as they memset  with `num_bytes` of `part2-part1` and later memcpy with `num_bytes` of `part4-part3`

Reply:
I'm not 100% certain it is a bug, since why would you need to memset if you're going to memcpy over the same amount of data? Of course, it could just be _overly_ defensive code.